modules/core/arrow-test/src/main/kotlin/arrow/test/laws/SemigroupalLaws.kt

```diff
@@ -0,0 +1,35 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.core.Tuple2
+import arrow.typeclasses.Eq
+import arrow.typeclasses.Semigroupal
+
+object SemigroupalLaws {
+
+    fun <F, A, B, C> laws(
+            SGAL: Semigroupal<F>,
+            af: Kind<F, A>,
+            bf: Kind<F, B>,
+            cf: Kind<F, C>,
```

Comment:
Instead of passing values here, we should have a constructor function even if that means only working with 1-2 concrete types such as `(Int) -> Kind<F, Int>` and `(String) -> Kind<F, String>`

Reply:
What would you think about passing a `Generator` instead of the 
constructor function? I am asking because currently this is just testing collections (SetK, ListK and SequenceK) with a single element.

Reply:
Keep it as a constructor for now

Reply:
Ok